types/index.d.ts

```diff
@@ -57,18 +57,19 @@ export class Type {
   compareBuffers(buf1: Buffer, buf2: Buffer): number;
   constructor(schema: Schema, opts?: any);
   createResolver(type: any, opts?: any): any;  // TODO: opts not documented on wiki
-  decode(buf: Buffer, pos?: number, resolver?: any): any;
-  encode(val: any, buf: Buffer, pos?: number): void; 
-  fingerprint(algorithm?: any): any;
+  decode(buf: Buffer, pos?: number, resolver?: any): { value: any, offset: number};
+  encode(val: any, buf: Buffer, pos?: number): void;
+  equals(type: Type): boolean;
+  fingerprint(algorithm?: string): Buffer | string;
```

Comment:
As far as I can see fingerprint can return Buffers only

Reply:
I think so too. I was deferring to https://nodejs.org/api/crypto.html#crypto_class_hash => https://nodejs.org/api/stream.html#stream_readable_read_size
but more than happy to change

Reply:
Let's change it to just `Buffer`.